lib/kube/proxy/resource_list.go

```diff
@@ -71,6 +75,49 @@ func (f *Forwarder) listResources(sess *clusterSession, w http.ResponseWriter, r
 		status = rw.Status()
 	} else {
 		allowedResources, deniedResources := sess.Checker.GetKubeResources(sess.kubeCluster)
+
+		details, err := f.findKubeDetailsByClusterName(sess.kubeClusterName)
+		if err != nil {
+			return nil, trace.Wrap(err, "failed to find kube details by cluster name %q", sess.kubeClusterName)
+		}
+		makeKind := func(resources []types.KubernetesResource) []types.KubernetesResource {
+			var out []types.KubernetesResource // Not pre-allocating as we don't know the final size. Likely 0 if no CRD resource defined.
+		loop:
+			for _, r := range resources {
+				tmp := strings.Split(r.Kind, "/")
+				if len(tmp) < 3 { // Expect <group>/<version>/<kind>. If not, skip.
+					continue
+				}
+				group, version, kind := tmp[0], tmp[1], strings.Join(tmp[2:], "/")
```

Comment:
There should always be exactly 3 parts here. The version and kind must be a valid domain name label as defined in RFC 1035, and the group must be a RFC 1123 subdomain.